### PR TITLE
 add paillier precompiled contract

### PR DESCRIPTION
**Pull request by Shareong**
header and source

### CODE REVIEW

File: libprecompiled/extension/PaillierPrecompiled.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    dev::eth::ContractABI abi;
+    bytes out;
+    std::string result;
+
+    if (func == name2Selector[PAILLIER_METHOD_SET_STR])
+    {
+        // paillierAdd(string,string)
+        std::string cipher1, cipher2;
+        abi.abiOut(data, cipher1, cipher2);
+        CallPaillier::Ptr callPaillier;
```
Comment (wheatli) [inline libprecompiled/extension/PaillierPrecompiled.cpp]: Can we set the callPaillier as a member variable? It will be with high performance if so.

Comment (Shareong) [reply libprecompiled/extension/PaillierPrecompiled.cpp]: Okay, that's a good suggestion.

File: libprecompiled/extension/PaillierPrecompiled.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+PaillierPrecompiled::PaillierPrecompiled()
+{
+    callPaillier = std::make_shared<CallPaillier>();
+    name2Selector[PAILLIER_METHOD_SET_STR] = getFuncSelector(PAILLIER_METHOD_SET_STR);
+}
+
+bytes PaillierPrecompiled::call(
+    ExecutiveContext::Ptr context, bytesConstRef param, Address const& origin)
+{
+    (void)(context), (void)(origin);
```
Comment (bxq2011hust) [inline libprecompiled/extension/PaillierPrecompiled.cpp]:
```suggestion
   
```

File: libprecompiled/extension/PaillierPrecompiled.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+const char* const PAILLIER_METHOD_SET_STR = "paillierAdd(string,string)";
+
+PaillierPrecompiled::PaillierPrecompiled()
+{
+    callPaillier = std::make_shared<CallPaillier>();
+    name2Selector[PAILLIER_METHOD_SET_STR] = getFuncSelector(PAILLIER_METHOD_SET_STR);
+}
+
+bytes PaillierPrecompiled::call(
+    ExecutiveContext::Ptr context, bytesConstRef param, Address const& origin)
```
Comment (bxq2011hust) [inline libprecompiled/extension/PaillierPrecompiled.cpp]:
```suggestion
    ExecutiveContext::Ptr, bytesConstRef param, Address const&)
```

File: libprecompiled/extension/PaillierPrecompiled.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+public:
+    typedef std::shared_ptr<PaillierPrecompiled> Ptr;
+    PaillierPrecompiled();
+    virtual ~PaillierPrecompiled(){};
+
+    virtual bytes call(dev::blockverifier::ExecutiveContext::Ptr context, bytesConstRef param,
+        Address const& origin = Address());
+
+private:
+    CallPaillier::Ptr callPaillier;
```
Comment (wheatli) [inline libprecompiled/extension/PaillierPrecompiled.h]: member variable should begin with m_

File: .gitignore [hunk 5]
```diff
@@ -36,6 +36,8 @@
 
 
 # ignore some directory
+cmake-build-debug
+cmake-build-debug/**
```
Comment (bxq2011hust) [inline .gitignore]: this is uncessary

**Pull request merged**